src/lustre/lustreArrayDependencies.ml

```diff
@@ -0,0 +1,252 @@
+(* This file is part of the Kind 2 model checker.
+
+   Copyright (c) 2020 by the Board of Trustees of the University of Iowa
+
+   Licensed under the Apache License, Version 2.0 (the "License"); you
+   may not use this file except in compliance with the License.  You
+   may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0 
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+   implied. See the License for the specific language governing
+   permissions and limitations under the License. 
+
+ *)
+
+(**
+    @author Andrew Marmaduke *)
+
+module A = LustreAst
+module AH = LustreAstHelpers
+module AIC = LustreAstInlineConstants
+module Ctx = TypeCheckerContext
+module Chk = LustreTypeChecker
+
+module StringMap = struct
+  include Map.Make(struct
+    type t = HString.t
+    let compare i1 i2 = HString.compare i1 i2
+  end)
+  let keys: 'a t -> key list = fun m -> List.map fst (bindings m)
+end
```

Comment:
I would replace this declaration with just: `module StringMap = HString.HStringMap`

The `keys` function is not really necessary (see my previous comment).

Reply:
My point was not only that the `keys` function could be removed but that doing so you could use directly the already defined types `HString.HStringMap` and `HString.HStringSet`. You only need to introduce an alias for convenience:

```
module StringMap = HString.HStringMap
module StringSet = HString.HStringSet
```